engine/notifications.js

```diff
@@ -0,0 +1,298 @@
+        );
+      }
+      // user decided to unregister from `all`
+      if (features.indexOf('all') >= 0) {
+        return redis.srem(client, 'all-notifications', deviceId)
+        .then(() => redis.srem(client, `${deviceId}-notifications`, 'all'));
+      }
+      // unregister from `all` and register to negative of `features`
+      return redis.srem(client, 'all-notifications', deviceId)
+      .then(() => redis.srem(client, `${deviceId}-notifications`, 'all'))
+      // register to everything except of features
+      .then(() => redis.get(client, 'status'))
+      .then(status => {
+        let allFeatures = ['new'];
+        allFeatures = allFeatures.concat(Object.keys(JSON.parse(status)));
+        return Promise.all(allFeatures.map(slug => {
+          if (features.indexOf(slug) >= 0) {
+            return Promise.resolve();
+          }
+          return redis.sadd(client, `${deviceId}-notifications`, slug)
+          .then(() => redis.sadd(client, `${slug}-notifications`, deviceId));
+        }));
+      });
+    });
+  })
+  .then(() => getRegisteredFeatures(deviceId, dbNumber))
+  .catch(err => {
+    if (err.message !== 'Not Found') {
+      throw err;
+    }
+    return [];
+  })
+  .then(regFeatures => {
+    if (!doNotConfirm && endpoint) {
+      return sendConfirmation(endpoint, key, deviceId, regFeatures, dbNumber)
+      .then(() => regFeatures);
+    }
+    return regFeatures;
+  });
+}
+
+// Registers to all notifications including new ones
+// Removes registrations to individual features
+function registerToAll(deviceId, dbNumber) {
+  return getRegisteredFeatures(deviceId, dbNumber)
+  .then(features => unregister(deviceId, features, dbNumber, true))
+  .then(() => Promise.all([
+    redis.sadd(client, 'all-notifications', deviceId),
+    redis.sadd(client, `${deviceId}-notifications`, 'all')]));
+}
+
+// registers to receive notifications
+// required info:
+// * deviceId
+// * list of feature slugs to register to
+//   special features are `new` and `all`
+//   * new - do notify about new features
+//   * all - all features including new, registering to all will remove
+//     all individual registrations
+// * endpoint [optional]
+// one can register to a feature without providing an endpoint
+// if it was already saved:
+// * key
+// * dbNumber
+function register(deviceId, features, endpoint, key, dbNumber) {
+  if (!features || features.length === 0) {
+    return Promise.reject(new Error('No features provided'));
+  }
+
+  return setClient(dbNumber)
+  .then(() => {
+    if (endpoint) {
+      const device = {
+        id: deviceId,
+        endpoint,
+        key: key || '',
+      };
+      redis.hmset(client, `device-${deviceId}`, device);
+      return device;
+    }
+    return redis.hgetall(client, `device-${deviceId}`);
+  })
+  .then(device => {
+    if (!device || !device.endpoint) {
+      throw new Error('No endpoint provided');
+    }
+  })
+  .then(() => {
+    if (features.indexOf('all') >= 0) {
```

Comment:
Yes, but more precisely would be
`new` - notify me that a new feature has been added